### ISSUE TITLE
Add checks in OneHotEncoding to ensure categorical pools have not changed

### ISSUE THREAD

**Issue by ablaom** [JuliaAI/MLJModels.jl#306]
```julia
using MLJ
X = (x = coerce(["black", "white", "white", "black"], Multiclass),)

model = OneHotEncoder()
mach = machine(model, X) |> fit!

julia> transform(mach, selectrows(X, 2:3))
(x__black = [0.0, 0.0],
 x__white = [1.0, 1.0],)

julia> Xproduction = (x = coerce(["white", "white"], Multiclass),)
(x = CategoricalArrays.CategoricalValue{String,UInt32}["white", "white"],)

julia> transform(mach, Xproduction)
(x__black = [1.0, 1.0],
 x__white = [0.0, 0.0],)
```

Wrong!

**Comment by ablaom** [comment]
Related: https://github.com/alan-turing-institute/MLJ.jl/issues/663

**Comment by ablaom** [comment]
Resolved by #360
